lib/msf/util/host.rb

```diff
@@ -52,7 +49,6 @@ def self.normalize_host(host)
         # to try, just log it and return what we were given
         if !norm_host
           dlog("Host could not be normalized: #{host.inspect}")
-          norm_host = host
```

Comment:
Sounds good, also not sure why you would assign this to a local variable that is being returned after deeming it invalid, should discard it as soon as possible so this makes sense from a design perspective as well.

My only concern though is that this will result in the library returning `nil`. Is this expected by code that uses this module or are we going to risk more `nil` errors? I agree this is the right way to do things though.

Reply:
Yup returning `nil` is expected in this case. `nil` is converted to NULL in the database as I understand it, which works since the field is marked as NULLABLE.